File: onlinebookstore/src/app/models/book.ts
import { Cart } from "./cart";
import { OrderItems } from "./order-items";


export class Books {
    id!: number;
    title!: string;
    authorName!: string;
    genre!: string;
    year!: number;
    price!: number;
    stockAvailable!: number;

    cart: Cart[] = [];
    orderItems: OrderItems[] = [];
}

File: src/main/java/com/cts/onlinebookstore/model/Book.java
package com.cts.onlinebookstore.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.HashSet;
import java.util.Set;

@Data
@Entity
@Table(name = "book")
public class
Book
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;
    private String title;
    private String authorName;
    private String genre;
    private int year;
    private double price;
    private int stockAvailable;

    @OneToMany(mappedBy = "book",cascade = CascadeType.ALL)
    private Set<Cart> cart = new HashSet<>();

    @OneToMany(mappedBy = "book",cascade = CascadeType.ALL)
    private Set<OrderItems> orderItems = new HashSet<>();
}
